processor/integrations/integrations.go

```diff
@@ -142,8 +144,23 @@ func GetPostInfo(transformRaw json.RawMessage) PostParameterT {
 	return postInfo
 }
 
+// GetUserIDForStreamDestination parses the payload to get userId
+func GetUserIDForStreamDestination(jsonData json.RawMessage) string {
+	var userID string
+	parsedJSON := gjson.ParseBytes(jsonData)
+	if parsedJSON.Get("output").Exists() {
+		parsedJSON = parsedJSON.Get("output")
+		userID = parsedJSON.Get("userId").Value().(string)
+	}
+	return userID
+}
+
 // GetUserIDFromTransformerResponse parses the payload to get userId
-func GetUserIDFromTransformerResponse(transformRaw json.RawMessage) string {
+func GetUserIDFromTransformerResponse(transformRaw json.RawMessage, destination string) string {
+
+	if misc.ContainsString(objectStreamDestination, destination) {
```

Comment:
This is the same as GetPostInfoNew().UserID


Now, we don't need to do the whole processing around version checks and GetPostInfoNew. Everyone would be on the latest transformer versions.

Remove all that and move the code of GetUserIDForStreamDestination into  GetUserIDFromTransformerResponse

Reply:
incorporated.

Reply:
3 our destinations(autopilot, segment, salesforce) in transformer still use version 0 of payload object. Should we keep it as is for now @SumanthPuram ?